src/jdk.jdwp.agent/share/native/libjdwp/threadControl.c

```diff
@@ -319,6 +319,7 @@ removeNode(ThreadList *list, ThreadNode *node)
     ThreadNode *prev;
     ThreadNode *next;
 
+    JDI_ASSERT(list == node->list);
```

Comment:
"list" argument is redundant
I suggest to drop it and replace this assert with
`ThreadList *list == node->list;`

Reply:
I was almost to suggest the same.

Reply:
Yes. I did the assert to first convince myself that this would always be the case. I was thinking it would be good to leave as-is until we get a full round of testing first. Are you ok if I file an RFE instead?

Reply:
There are only 2 calls of removeNode and I don't see risk here. But fell free to file separate RFE

Reply:
I agree with Alex that there should not be any risk here.
But I'm okay if filing a separate RFE is more convenient to you.

Reply:
I filed [JDK-8295810](https://bugs.openjdk.org/browse/JDK-8295810). Thanks for the reviews!